pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,334 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+
+import re
+import time
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_range
+from pymeasure.instruments.validators import strict_discrete_set
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class EurotestHPP120256(Instrument):
+    """ Represents the Euro Test High Voltage DC Source model HPP-120-256
+    and provides a high-level for interacting with the instrument using the
+    Euro Test command set (Not SCPI command set).
+
+    .. code-block:: python
+
+    hpp120256 = EurotestHPP120256("GPIB0::20::INSTR",includeSCPI=False)
+
+    response = hpp120256.id
+    print(response)
+    print(hpp120256.lam_status)
+    print(hpp120256.status)
+
+    hpp120256.ramp_to_zero(100.0)
+
+    hpp120256.voltage_ramp = 50.0  # V/s
+    time.sleep(command_delay)
```

Comment:
It seems to me that this delay is needed after sending any command.
A possible solution, more user friendly, could be that you record the time when last `write` was executed and when a new `write` is requested, you check that at least command_delay has expired, otherwise wait before executing the write.
Of course, this would require to override `write` method

Reply:
As in Intrument ask method where is used time.sleep with a query_delay, here in write method could be possible the use of a delay_after_write. For me could be better leave the query_delay=0 and make use of that  new delay as I need both. What do you think? Thanks.

Note: I think remove change query_delay in ask and leave that new delay un write is not an option.

Reply:
I don't understand, whether you want a delay after write or not.

Nevertheless, if you consider it opportune, you can subclass `write` which calls afterwards `wait_for` (which calls sleep). With a parameter you may distinguish between cases, where you need it and those where not (I'd default to not wait).

Reply:
Yes.I need it because the instrument (old) miss commands if we do not wait between two consecutive writes. Some occurs in the ask case (write and read without waiting in the middle). I will considers your proposal. Thanks.

Reply:
I would suggest to override write and add the delay you need.

Reply:
Done in commit. Thanks